src/App.tsx

```diff
@@ -119,46 +125,82 @@ function App() {
       <BoxBasic>
         <TableContainer component={Paper}>
           <Table sx={{ minWidth: 650 }} aria-label="simple table">
+            <TableHead>
+              <TableRow>
+                <TableCell>Timestamp</TableCell>
+                <TableCell>Debugging Level</TableCell>
+                <TableCell>Source</TableCell>
+                <TableCell>Log Messages</TableCell>
+              </TableRow>
+            </TableHead>
             <TableBody>
-              {logs.map((log) => (
-                <TableRow
-                  key={log}
-                  sx={{ '&:last-child td, &:last-child th': { border: 0 } }}
-                >
-                  <TableCell component="th" scope="row">
-                    {log}
-                  </TableCell>
+              {logs.map((log,index) => (
+                <TableRow>                  
+                  <TableCell>{time[index]}</TableCell>
+                  <TableCell>{debug[index]}</TableCell>
+                  <TableCell>{source[index]}</TableCell>
+                  <TableCell>{log}</TableCell>
+                  {/* sx={{ '&:last-child td, &:last-child th': { border: 0 } }} */}
                 </TableRow>
               ))}
-            </TableBody>
+            </TableBody>            
           </Table>
         </TableContainer>
       </BoxBasic>
     </ThemeProvider>
   );
 }
 
-function getMessage(logging: JSON): undefined | string[] {
+function getMessage(logging: JSON): undefined | string[][] {
   const data = JSON.parse(JSON.stringify(logging));
   for (const key in data.results) {
     if ("search_types" in data.results[key]) {
       const id = data.results[key].search_types;
       const message: string[] = [];
+      const timestamp: string[] =[];
+      const source: string[] =[];
+      const debug: string[] =[];
       for (const keys in id) {
         if ("messages" in id[keys]) {
           const logs = id[keys].messages;
+          // populate different components of logged data and identifying log level
           for (const msg in logs) {
-            message.push(
-              `${logs[msg]["message"]["timestamp"]} ${logs[msg]["message"]["source"]} ${logs[msg]["message"]["message"]}`
+            const formattedTimestamp = logs[msg]["message"]["timestamp"].replace(/[TZ]/g, ' ')
+            timestamp.push(
+              `${formattedTimestamp}`
             );
+            source.push(
+              `${logs[msg]["message"]["source"]}`
+            )
+            const text = logs[msg]["message"]["message"];
+            const [debug_level, log_message] = debugLevel(text);
+            debug.push(debug_level);
+            message.push(log_message);
+
           }
-          return message;
+          return [timestamp,source,debug,message];
         }
       }
     }
   }
 }
 
+function debugLevel(text: string): [string, string] {
```

Comment:
I think this should be called `logLevel`. Also, is there really nothing about this in the data structure received from graylog? I would hope we would get something like that rather than having to scrape it out of the message text.

Reply:
On this, there is a log level, which gives an integer, we could use the number and scrape out the level if that would be a better way? What do you think about this?